ext/dom/xpath_rust.h

```diff
@@ -0,0 +1 @@
+extern char* domxpath_quote_literal(const char *const input, uintptr_t *const len);
```

Comment:
The choice of `uintptr_t` looks odd for a size.

Reply:
yeah, that's adopted from rust's `usize`, [quote](https://doc.rust-lang.org/std/primitive.usize.html)
> Primitive Type [usize](https://doc.rust-lang.org/std/primitive.usize.html#): The pointer-sized unsigned integer type.

php-src's `uintptr_t` should be equivalent to rust's `usize`

Reply:
> php-src's `uintptr_t` should be equivalent to rust's `usize`

Nope. `size_t` is the type to use for sizes.

Reply:
Okay, apparently Rust's `usize` maps to `uintptr_t` indeed, but this isn't really accurate from a C perspective :thinking: I expect both types to be the same on modern platforms, but it bothers me.

Reply:
oh damn there actually is a `core::ffi::c_size_t` in Rust-nightly, but it's not released/stable yet, quote:
>🔬This is a nightly-only experimental API. (c_size_t [#88345](https://github.com/rust-lang/rust/issues/88345))

Reply:
I would assume we'd depend on `libc` at some point anyway, so `libc::size_t` can probably be used.